protocol/src/debugProtocol.ts

```diff
@@ -411,6 +411,26 @@ export module DebugProtocol {
 	export interface RestartFrameResponse extends Response {
 	}
 
+	/** Goto request; value of command field is "goto".
+		The request sets the location where the debuggee will continue to run.
+		This makes it possible to skip the execution of code or to executed code again.
+		The code between the current location and the goto target is not executed but skipped.
+		The debug adapter first sends the GotoResponse and then a StoppedEvent (event type 'goto').
+	*/
+	export interface GotoRequest extends Request {
+		arguments: GotoArguments;
+	}
+	/** Arguments for "goto" request. */
+	export interface GotoArguments {
+		/** Set the goto target for this thread. */
+		threadId: number;
+		/** The location where the debuggee will continue to run. */
+		targetId: number;
+	}
+	/** Response to "goto" request. This is just an acknowledgement, so no body field is required. */
+	export interface GotoResponse extends Response {
+	}
```

Comment:
@jacdavis the current proposal cannot handle that case because the `GotoRequest` takes an ID of a target returned from the `GotoTargetsRequest`. You cannot manufacture these targets from a cursor location.
From your initial request I was assuming that all targets have to be returned from `GotoTargetsRequest`.